mlflow/store/rest_store.py

```diff
@@ -148,7 +148,8 @@ def create_run(self, experiment_id, user_id, run_name, source_type, source_name,
             start_time=start_time, source_version=source_version, tags=tag_protos))
         response_proto = self._call_endpoint(CreateRun, req_body)
         run = Run.from_proto(response_proto.run)
-        self.set_tag(run.info.run_uuid, RunTag(key=MLFLOW_RUN_NAME, value=run_name))
+        if run_name:
+            self.set_tag(run.info.run_uuid, RunTag(key=MLFLOW_RUN_NAME, value=run_name))
```

Comment:
Why do file_store and rest_store have different behaviors here?
FileStore does:

```
        self.set_tag(run_uuid, RunTag(
            key=MLFLOW_RUN_NAME, value=run_name or "Run %s" % num_runs))
```

Reply:
Good question, the FileStore default seems like an artifact of the initial release / not necessary - removed it in this PR so that the two stores behave the same. With this change users will see a run name of "Run <some-uuid>" in the UI instead of "Run 1", "Run 2", etc when creating runs against the FileStore, but that doesn't seem too bad